### PR TITLE
add type=module to addScriptTag

### LINKED ISSUE #359: Add type=module to addScriptTag
https://github.com/kblok/puppeteer/pull/2/commits/625c7ebdda2ef0473ee8d593dd30a1d34d3ee54a

### PR DESCRIPTION
**Pull request by Meir017**
closes #359

### CODE REVIEW

**Comment by Meir017** [comment]
these tests are failing
* ShouldWorkWithAPath
* ShouldIncludeSourcemapWhenPathIsProvided
* ShouldWorkWithContent

looks like an issue with the new `type` property on the `AddTagOptions` class

File: lib/PuppeteerSharp/Frame.cs [hunk 1]
```diff
@@ -406,31 +406,40 @@ public Task<string[]> SelectAsync(string selector, params string[] values)
...
               document.head.appendChild(script);
               return script;
             }";
 
+            async Task<ElementHandle> AddScriptTagPrivate(string script, string urlOrContent, string type)
+            {
+                var context = await GetExecutionContextAsync();
+                return string.IsNullOrEmpty(type)
+                        ? (await context.EvaluateFunctionHandleAsync(script, urlOrContent)) as ElementHandle
+                        : (await context.EvaluateFunctionHandleAsync(script, urlOrContent, type)) as ElementHandle;
```
Comment (kblok) [inline lib/PuppeteerSharp/Frame.cs]: Can't we do this?

```cs 
return await (string.IsNullOrEmpty(type)
    ? context.EvaluateFunctionHandleAsync(script, urlOrContent)
    : context.EvaluateFunctionHandleAsync(script, urlOrContent, type)) as ElementHandle;
```

File: lib/PuppeteerSharp/Frame.cs [hunk 2]
```diff
@@ -406,31 +406,40 @@ public Task<string[]> SelectAsync(string selector, params string[] values)
...
+            const string addScriptContent = @"function addScriptContent(content, type = 'text/javascript') {
               const script = document.createElement('script');
-              script.type = 'text/javascript';
+              script.type = type;
               script.text = content;
               document.head.appendChild(script);
               return script;
             }";
 
+            async Task<ElementHandle> AddScriptTagPrivate(string script, string urlOrContent, string type)
```
Comment (kblok) [inline lib/PuppeteerSharp/Frame.cs]: Let's add `private` here

Comment (kblok) [reply lib/PuppeteerSharp/Frame.cs]: I don't like much the `Private` suffix here. This could be a private override right?

Comment (Meir017) [reply lib/PuppeteerSharp/Frame.cs]: It's a local method

Comment (kblok) [reply lib/PuppeteerSharp/Frame.cs]: 🤓

**Pull request merged**